docs/topics/multiplatform/multiplatform-compatibility-guide.md

```diff
@@ -852,3 +852,32 @@ and prepare this functionality for the upcoming Kotlin 2.0 release. From now on:
 
 Starting with Kotlin 1.9.20, you need to explicitly make a cast to and from the corresponding C and Objective-C forward
 declarations. Also, it's now only possible to import forward declarations by using special packages.
+
+## Incorrect version of iOS framework after Kotlin upgrade
+
+**What's the issue?**
+
+If you use direct integration through the `embedAndSignAppleFrameworkForXcode` task, which connects the iOS framework
+from your multiplatform project to the iOS app in Xcode, you might encounter an issue when changing the Kotlin version.
+
+If you upgrade the Kotlin version from 1.9.2x to 2.0.0 in your multiplatform project (or downgrade it from 2.0.0 to 1.9.2x),
+then make changes in Kotlin files and try building the app, Xcode may incorrectly use the previous version of the iOS
+framework. So, the changes won't be visible in the iOS app in Xcode.
+
+**What's the workaround?**
+
+1. In Xcode, clean build directories using **Product** | **Clean Build Folder**.
+2. In the terminal, run the following command:
+
+   ```none
+   ./gradlew clean
+   ```
+
+3. Build the app again to ensure that the new version of the iOS framework is used. 
+
+**When will the issue be fixed?**
+
+We're planning to fix this issue in Kotlin 2.0.10 and later versions. You can check if any preview versions of
```

Comment:
```suggestion
We're planning to fix this issue in Kotlin 2.0.10. You can check if any preview versions of
```

Isn't it implicit that if we fix it in 2.0.10, we count on it to be fixed later on?